app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt

```diff
@@ -398,80 +398,62 @@ class MarkChaptersCompletedFragmentTest {
     launch<MarkChaptersCompletedTestActivity>(
       createMarkChaptersCompletedTestActivityIntent(internalProfileId)
     ).use {
-      testCoroutineDispatchers.runCurrent()
-      onView(withId(R.id.mark_chapters_completed_all_check_box_container)).perform(click())
-      scrollToPosition(position = 1)
-      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 1)
-      scrollToPosition(position = 2)
-      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 2)
-      scrollToPosition(position = 3)
-      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 3)
-      scrollToPosition(position = 5)
-      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 5)
-      scrollToPosition(position = 7)
-      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 7)
-      scrollToPosition(position = 8)
-      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 8)
-      scrollToPosition(position = 10)
-      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 10)
-      scrollToPosition(position = 11)
-      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 11)
-      scrollToPosition(position = 13)
-      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 13)
-      scrollToPosition(position = 14)
-      verifyItemCheckedOnRecyclerViewItemAtPosition(itemPosition = 14)
+      testMarkChaptersCompleted_selectAll_selectsAllTests()
     }
   }
 
+  private fun testMarkChaptersCompleted_selectAll_selectsAllTests() {
```

Comment:
This wasn't what I meant, actually. My earlier comment was suggesting to:
- Move the 'scrollToPosition' call to be inside of 'verifyItemCheckedOnRecyclerViewItemAtPosition' so that we only need 1 line of code in the test to perform _each_ item interaction (not all of them)
- It's an antipattern to move the behavior-specific steps of a test into a helper as it hides away the thing that's actually being tested (which can make the tests harder to understand at best, and difficult to maintain at worst in the case where the helper is used by multiple tests).

Please revert and make the mentioned adjustments above. If you're not sure, feel free to reach out.

Reply:
Hey @BenHenning i have updated and moved 'scrollToPosition' also have reverted some changes.